sampleSpecs/refined.smithy

```diff
@@ -5,18 +5,41 @@ use smithy4s.meta#refined
 @trait(selector: "integer")
 @refined(
   targetClasspath: "smithy4s.example.refined.Age",
-  providerClasspath: "smithy4s.example.refined.Age.provider",
-  canonicalShape: Age
+  providerClasspath: "smithy4s.example.refined.Age.provider"
 )
 structure ageFormat {}
 
+@trait(selector: "list")
+@refined(
```

Comment:
let's call it "refinement", because it's not the trait that is refined, it's the type that the trait ends up being applied on.

Reply:
Addressed in [aad9b81](https://github.com/disneystreaming/smithy4s/pull/333/commits/aad9b813d4ab58e98e89aa63095574d1baa1e313)